autoPyTorch/pipeline/components/setup/network_head/fully_convolutional.py

```diff
@@ -70,32 +71,41 @@ def get_properties(dataset_properties: Optional[Dict[str, Any]] = None) -> Dict[
         }
 
     @staticmethod
-    def get_hyperparameter_search_space(dataset_properties: Optional[Dict[str, str]] = None,
-                                        min_num_layers: int = 1,
-                                        max_num_layers: int = 4,
-                                        min_num_filters: int = 16,
-                                        max_num_filters: int = 256) -> ConfigurationSpace:
+    def get_hyperparameter_search_space(
+        dataset_properties: Optional[Dict[str, str]] = None,
+        num_layers: HyperparameterSearchSpace = HyperparameterSearchSpace(hyperparameter="num_layers",
+                                                                          value_range=(1, 4),
+                                                                          default_value=2),
+        num_filters: HyperparameterSearchSpace = HyperparameterSearchSpace(hyperparameter="num_filters",
+                                                                           value_range=(16, 256),
+                                                                           default_value=32),
+        activation: HyperparameterSearchSpace = HyperparameterSearchSpace(hyperparameter="activation",
+                                                                          value_range=tuple(_activations.keys()),
+                                                                          default_value=list(_activations.keys())[0]),
+        pooling_method: HyperparameterSearchSpace = HyperparameterSearchSpace(hyperparameter="pooling_method",
+                                                                              value_range=("average", "max"),
+                                                                              default_value="max"),
+    ) -> ConfigurationSpace:
         cs = ConfigurationSpace()
 
-        num_layers_hp = UniformIntegerHyperparameter("num_layers",
-                                                     lower=min_num_layers,
-                                                     upper=max_num_layers)
+        min_num_layers, max_num_layers = num_layers.value_range
+        num_layers_hp = get_hyperparameter(num_layers, UniformIntegerHyperparameter)
 
-        pooling_method_hp = CategoricalHyperparameter("pooling_method",
-                                                      choices=["average", "max"])
+        add_hyperparameter(cs, pooling_method, CategoricalHyperparameter)
 
-        activation_hp = CategoricalHyperparameter('activation',
-                                                  choices=list(_activations.keys()))
+        activation_hp = get_hyperparameter(activation, CategoricalHyperparameter)
 
-        cs.add_hyperparameters([num_layers_hp, pooling_method_hp, activation_hp])
```

Comment:
this pooling_method_hp was extra?

Reply:
no it has been added in line 94. For hyperparameters that have a condition on them, we can't directly add them to the config space, we need the variable for that hyperparameter and that's why num_layers_hp and activation_hp are explicitly added